src/__tests__/commands/devices/preferences.test.ts

```diff
@@ -0,0 +1,154 @@
+import type { CommandArgs } from '../../../commands/apps.js'
+import type { BuildOutputFormatterFlags } from '../.././../lib/command/output-builder.js'
+import type { SmartThingsCommandFlags } from '../../../lib/command/smartthings-command.js'
+import type { ChooseFunction } from '../../../lib/command/util/util-util.js'
+import type { Table, TableGenerator } from '../../../lib/table-generator.js'
+import { buildArgvMock, buildArgvMockStub } from '../../test-lib/builder-mock.js'
+
+
+const apiCommandMock = jest.fn<typeof apiCommand>()
+const apiCommandBuilderMock = jest.fn<typeof apiCommandBuilder>()
+jest.unstable_mockModule('../../../lib/command/api-command.js', () => ({
+	apiCommand: apiCommandMock,
+	apiCommandBuilder: apiCommandBuilderMock,
+}))
+
+const chooseDeviceMock = jest.fn<ChooseFunction<Device>>()
+jest.unstable_mockModule('../../../lib/command/util/devices-util.js', () => ({
+	chooseDevice: chooseDeviceMock,
+}))
+
+const formatAndWriteItemMock = jest.fn<typeof formatAndWriteItem<DevicePreferenceResponse>>()
+const formatAndWriteItemBuilderMock = jest.fn<typeof formatAndWriteItemBuilder>()
+jest.unstable_mockModule('../../../lib/command/format.js', () => ({
+	formatAndWriteItem: formatAndWriteItemMock,
+	formatAndWriteItemBuilder: formatAndWriteItemBuilderMock,
+}))
+
+
+const { default: cmd, buildTableOutput } = await import('../../../commands/devices/preferences.js')
+
+
+test('builder', () => {
+	const yargsMock = buildArgvMockStub<object>()
+	const {
+		yargsMock: apiCommandBuilderArgvMock,
+		positionalMock,
+		exampleMock,
+		argvMock,
+	} = buildArgvMock<SmartThingsCommandFlags, BuildOutputFormatterFlags>()
+
+	apiCommandBuilderMock.mockReturnValue(apiCommandBuilderArgvMock)
+	formatAndWriteItemBuilderMock.mockReturnValue(argvMock)
+
+	const builder = cmd.builder as (yargs: Argv<object>) => Argv<CommandArgs>
+
+	expect(builder(yargsMock)).toBe(argvMock)
+
+	expect(apiCommandBuilderMock).toHaveBeenCalledExactlyOnceWith(yargsMock)
+	expect(formatAndWriteItemBuilderMock).toHaveBeenCalledExactlyOnceWith(apiCommandBuilderArgvMock)
+	expect(positionalMock).toHaveBeenCalledOnce()
+	expect(exampleMock).toHaveBeenCalledOnce()
+})
+
+const tablePushMock = jest.fn<Table['push']>()
+const tableToStringMock = jest.fn<Table['toString']>().mockReturnValue('table output')
+const tableMock: Table = {
+	push: tablePushMock,
+	toString: tableToStringMock,
+}
+const newOutputTableMock = jest.fn<TableGenerator['newOutputTable']>().mockReturnValue(tableMock)
+const tableGeneratorMock = { newOutputTable: newOutputTableMock } as unknown as TableGenerator
+
+const preferences: DevicePreferenceResponse = {
+	values: {
+		key0: {
+			preferenceType: 'string',
+			value: 'preference string 0',
+		},
+		key2: {
+			preferenceType: 'string',
+			value: 'preference string 2',
+		},
+		key1: {
+			preferenceType: 'string',
+			value: 'preference string 1',
+		},
+	},
+}
+
+describe('buildTableOutput', () => {
+	it('works', () => {
```

Comment:
Is this test left over from an initial stub or is it still useful?

Reply:
Oops, good catch. I'll remove this in the next pull request.